warehouse/models/mart/benefits/fct_benefits_events.sql

```diff
@@ -41,10 +80,42 @@ WITH fct_benefits_events AS (
         {{ json_extract_column('event_properties', 'language') }},
         {{ json_extract_column('event_properties', 'origin') }},
         {{ json_extract_column('event_properties', 'path') }},
-        {{ json_extract_column('event_properties', 'payment_group') }},
         {{ json_extract_column('event_properties', 'status') }},
         {{ json_extract_column('event_properties', 'transit_agency') }},
-        {{ json_extract_flattened_column('event_properties', 'eligibility_types') }},
+
+        -- New column `enrollment_method`, historical values should be set to "digital"
+        -- https://github.com/cal-itp/benefits/pull/2402
+        COALESCE(
+          {{ json_extract_column('event_properties', 'enrollment_method', no_alias = true) }},
+          "digital"
+        ) AS event_properties_enrollment_method,
+
+        -- Historical data existed in `auth_provider` but new data is in `claims_provider`
+        -- https://github.com/cal-itp/benefits/pull/2401
+        COALESCE(
+          {{ json_extract_column('event_properties', 'claims_provider', no_alias = true) }},
+          {{ json_extract_column('event_properties', 'auth_provider', no_alias = true) }}
+        ) AS event_properties_claims_provider,
+        -- include the old field as well, deprecated in the mart definition
+        event_properties_claims_provider AS event_properties_auth_provider,
```

Comment:
This pattern doesn't work in dbt, it produces an error:
```
(poetry_env) VevePro:warehouse vivek$ poetry run dbt run -s +fct_benefits_events
19:14:12  Running with dbt=1.5.1
19:14:16  [WARNING]: Configuration paths exist in your dbt_project.yml file which do not apply to any resources.
There are 1 unused configuration paths:
- models.calitp_warehouse.mart.ad_hoc
19:14:16  Found 420 models, 950 tests, 0 snapshots, 0 analyses, 852 macros, 0 operations, 12 seed files, 175 sources, 4 exposures, 0 metrics, 0 groups
19:14:16
19:14:21  Concurrency: 8 threads (target='dev')
19:14:21
19:14:21  1 of 2 START sql view model vb_staging.stg_amplitude__benefits_events .......... [RUN]
19:14:22  1 of 2 OK created sql view model vb_staging.stg_amplitude__benefits_events ..... [CREATE VIEW (0 processed) in 1.36s]
19:14:22  2 of 2 START sql table model vb_mart_benefits.fct_benefits_events .............. [RUN]
19:14:25  BigQuery adapter: https://console.cloud.google.com/bigquery?project=cal-itp-data-infra-staging&j=bq:us-west2:e2a122f9-54d8-4065-8ab3-43c72852c320&page=queryresults
19:14:25  2 of 2 ERROR creating sql table model vb_mart_benefits.fct_benefits_events ..... [ERROR in 2.96s]
19:14:25
19:14:25  Finished running 1 view model, 1 table model in 0 hours 0 minutes and 8.80 seconds (8.80s).
19:14:25
19:14:25  Completed with 1 error and 0 warnings:
19:14:25
19:14:25  Database Error in model fct_benefits_events (models/mart/benefits/fct_benefits_events.sql)
19:14:25    Unrecognized name: event_properties_claims_provider at [158:9]
19:14:25    compiled Code at target/run/calitp_warehouse/models/mart/benefits/fct_benefits_events.sql
19:14:25
19:14:25  Done. PASS=1 WARN=0 ERROR=1 SKIP=0 TOTAL=2
```
I believe other instances of this pattern will fail as well.

Reply:
Fixed in [83ff624](https://github.com/cal-itp/data-infra/pull/3468/commits/83ff624563f22a9cd790f4bb64f4ac4d54ff34d7)